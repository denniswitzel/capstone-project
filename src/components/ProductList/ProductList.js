import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import styled from 'styled-components'
import Filter from '../Filter/Filter'
import Header from '../Header/Header'
import Product from '../Products/Product'
import PropTypes from 'prop-types'

ProductList.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  category: PropTypes.string
}

export default function ProductList({ product, headline }) {
  const filters = ['Show all', 'gluten', 'soy']
  const [active, setActive] = useState(filters[0])
  const [filterToggle, setFilterToggle] = useState(false)

  const history = useHistory()
  
  return (
    <ProductListWrapper>
      <Header headline={headline} active={filterToggle.toString()} onToggle={() => setFilterToggle(!filterToggle)}/>
      {filterToggle &&
      <Filter filters={filters} active={active} setActive={setActive}/>}
      {product?.filter(product => active === 'Show all' || !product.allergies.includes(active)).map((productItem) => (
        <Product
          onClick={() => history.push(`/${productItem.category}/${productItem.id}`)}
          key={productItem.id}
          title={productItem.title}
          image={productItem.image}
        />
      ))}
    </ProductListWrapper>
  )
}

const ProductListWrapper = styled.section`
    grid-column: 1/3;
    padding-bottom: 120px;
    max-width: 400px;
`