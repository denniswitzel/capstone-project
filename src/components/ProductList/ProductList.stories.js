import React from 'react'
import ProductList from './ProductList'
import RatingList from '../RatingList/RatingList'

export default {
  title: 'Pages/Product List',
  component: ProductList,
  argTypes: {
    headline: { control: {
      type: 'text',
      options: 'Burger',
    }}
  },
}


const Template = (args) => 
  <ProductList {...args}/>

const Rating = (args) => 
  <RatingList {...args}/>

export const multipleProducts = Template.bind({})
multipleProducts.args = {
  headline: 'Burger',
  product: [
    {
      id: 1,
      title: 'amidori - veggie best burger',
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Amidori-Veggie-Best-Burger.png',
      gluten: ['gluten'],
    },
    {
      id: 2,
      title: 'followfood - FREI!BURGER',
      ingredients:
        'Seitan* 54 % (Wasser, Weizeneiweiß), Weizeneiweißtexturat 26 % (Weizeneiweiß, Weizenmehl), ölsäurereiches (High-Oleic) Sonnenblumenöl, Gewürze, Sojasauce (Wasser, Sojabohnen, Meersalz), Hefeextrakt, Zwiebeln, Reismehl, Verdickungsmittel: Johannisbrotkernmehl',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Followfood-Freiburger.png',
    },
    {
      id: 3,
      title: 'Garden Gourmet - Sensational Burger',
      ingredients:
        'Trinkwasser, 18,5 % SOJAPROTEIN, pflanzliche Öle (Raps, Kokosnuss), natürliche Aromen, 1,7% WEIZENPROTEIN, Stabilisator (Methylcellulose), Branntweinessig, Frucht- und Pflanzenkonzentrate (Rote Beete, Paprika, Karotte, schwarze Johannisbeere), Speisesalz, GERSTENMALZEXTRAKT',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Gourmet-Garden-Vegane-Burger.png',
    },
    {
      id: 4,
      title: 'Rügenwalder - Mühlen Burger',
      ingredients:
        'Trinkwasser, 16% Sojaproteinkonzentrat, Rapsöl, Zwiebeln, Weizengluten, Kochsalz, Branntweinessig, Verdickungsmittel: Methylcellulose, Aroma, Weizenstärke, färbende Lebensmittel: Konzentrat aus Rote Beete, Karotte, Rettich',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Ruegenwalder-Muehlen-Burger.png',
    },
    {
      id: 5,
      title: 'Rügenwalder - Mühlen Crispyburger',
      ingredients:
        'Trinkwasser, Weizenmehl, 11% Sojaproteinkonzentrat, Weizengluten, Rapsöl, 4% Sojaproteinisolat, Kochsalz, Verdickungsmittel: Methylcellulose, Branntweinessig, Maismehl, natürliches Aroma, Weizenstärke, Zucker, Gewürze, Traubenzucker, Hefe',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Ruegenwalder-Muehlen-Crispyburger.png',
    },
    {
      id: 6,
      title: 'The Vegetarian Butcher - Sieht-Chick-Aus Burger',
      ingredients:
        '63% Sojabasis (Wasser, texturiertes SOJAEIWEISS, WEIZENSTÄRKE, WEIZENEIWEISS), dicke Bohnen, Sonnenblumenöl, Palmöl, Gewürzextrakte (Koriander, Nelke, Zwiebel, Oregano), Verdickungsmittel (Methylcellulose), natürliches Aroma, Bambusfaser, Stärke, Citrusfaser, Gewürze (Muskatblüte, Ingwer, Knoblauch, Pfeffer, Zwiebel), Speisesalz, karamellisierter Zucker, Stabilisator (verarbeitete Euchema-Algen), Eisendiphosphat, Vitamin B12',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/The-Vegetarian-Butcher-Vegane-Sieht-Chick-Aus-Burger.png',
    },
    {
      id: 7,
      title: 'Vivera - Veganer Burger',
      ingredients:
        '47% Kürbis*, 29% Süßkartoffel*, Sonnenblumenöl, SOJA-EIWEIß, Kartoffelstärke, Gewürze, Verdickungsmittel (Methylcellulose), Essig, WEIZENFASERN, Salz, rote Paprika, Wasser, Zwiebel, natürliche Aromen, Knoblauch',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Vivera-Veganer-Burger.png',
    },
    {
      id: 8,
      title: 'Wheaty - Vegan Superhero Burger',
      ingredients:
        'Seitan* (Wasser, WEIZENeiweiß*) (54%), WEIZENeiweißtexturat* (WEIZENeiweiß*, WEIZENmehl*) ölsäurereiches (High-Oleic) Sonnenblumenöl*, Zwiebeln*, SOJAsauce* (Wasser, SOJAbohnen*, Meersalz), Hefeextrakt*, Gewürze*, Reismehl*, Verdickungsmittel: Johannisbrotkernmehl',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Wheat-Vegan-Superhero-Burger.png',
    },
    {
      id: 9,
      title: 'Wheaty - Veganer Tex-Mex Burger',
      ingredients:
        'Seitan* (Wasser, WEIZENeiweiß*) (42%), WEIZENeiweißtexturat* (WEIZENeiweiß*, WEIZENmehl*) (13%), Paprika rot*, ölsäurereiches (High-Oleic) Sonnenblumenöl*, SOJAsauce* (Wasser, SOJAbohnen*, Meersalz), Zwiebeln*, Knoblauch*, Gewürze*, Steinsalz, Reismehl*, Verdickungsmittel: Johannisbrotkernmehl',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Wheat-Vegan-Tex-Mex-Burger.png',
    },
  ],
}

export const singleProducts = Template.bind({})
singleProducts.args = {
  headline: 'Burger',
  product: [
    {
      id: 1,
      title: 'amidori - veggie best burger',
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Amidori-Veggie-Best-Burger.png',
      gluten: ['gluten'],
    }],
}

export const ratedProducts = Rating.bind({})
ratedProducts.args = {
  headline: 'Burger',
  product: [
    {
      id: 1,
      title: 'amidori - veggie best burger',
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Amidori-Veggie-Best-Burger.png',
      gluten: ['gluten'],
      rating: [{session_id: "1234", rating:
      5}, {rating: 4}]
    },{
      id: 2,
      title: 'followfood - FREI!BURGER',
      ingredients:
        'Seitan* 54 % (Wasser, Weizeneiweiß), Weizeneiweißtexturat 26 % (Weizeneiweiß, Weizenmehl), ölsäurereiches (High-Oleic) Sonnenblumenöl, Gewürze, Sojasauce (Wasser, Sojabohnen, Meersalz), Hefeextrakt, Zwiebeln, Reismehl, Verdickungsmittel: Johannisbrotkernmehl',
      allergies: ['Gluten', 'Soy'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Followfood-Freiburger.png',
      rating: [{session_id: "1234", rating:
      4}, {rating: 4}]
    }],
}