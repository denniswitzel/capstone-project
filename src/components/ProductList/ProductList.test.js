import React from 'react'
import renderer from 'react-test-renderer'
import ProductList from './ProductList'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'

describe('ProductList', () => {
  const headline = 'BURGER'
  const product = [
    {
      id: 1,
      title: 'amidori - veggie best burger',
      ingredients:
        'Wasser, Erbsenproteinisolat 17 %, Zwiebelwürfel, Rapsöl, Kokosfett, Branntweinessig, Verdickungsmittel Methylcellulose, Tomatenmark, Apfelessig, Malzextrakt (Wasser, glutenfreies GERSTENRÖSTMALZ, Hopfen, Hefe), Citrusfaser, Erbsenfaser, glutenfreies HAFERVOLLKORNMEHL, Kartoffelstärke, Gewürze, Speisesalz, färbendes Frucht- und Gemüsekonzentrat (Hibiskus, Karotte, rote Bete), Hefeextrakt, natürliches Aroma, Antioxidationsmittel: Extrakt aus Rosmarin, Rauch',
      allergies: ['Gluten'],
      image:
        'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Amidori-Veggie-Best-Burger.png',
    },
  ]

  it('display product, category and headline', () => {
    const { getByText } = render(
      <MemoryRouter>
        <ProductList
          product={product}
          headline={headline}
        />
      </MemoryRouter>
    )
    expect([
      {
        id: 1,
        title: 'amidori - veggie best burger',
        ingredients:
          'Wasser, Erbsenproteinisolat 17 %, Zwiebelwürfel, Rapsöl, Kokosfett, Branntweinessig, Verdickungsmittel Methylcellulose, Tomatenmark, Apfelessig, Malzextrakt (Wasser, glutenfreies GERSTENRÖSTMALZ, Hopfen, Hefe), Citrusfaser, Erbsenfaser, glutenfreies HAFERVOLLKORNMEHL, Kartoffelstärke, Gewürze, Speisesalz, färbendes Frucht- und Gemüsekonzentrat (Hibiskus, Karotte, rote Bete), Hefeextrakt, natürliches Aroma, Antioxidationsmittel: Extrakt aus Rosmarin, Rauch',
        allergies: ['Gluten'],
        image:
          'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/burger/Amidori-Veggie-Best-Burger.png',
      },
    ]).toEqual(expect.arrayContaining(product))
    expect(getByText(headline)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <ProductList />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
