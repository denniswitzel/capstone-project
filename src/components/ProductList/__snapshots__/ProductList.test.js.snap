// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<section
  className="sc-fzplWN ehPxTb"
>
  <section
    className="sc-AxiKw dmdrCS"
  >
    <svg
      className="sc-AxhUy gikrmK"
      onClick={[Function]}
    >
      back-arrow.svg
    </svg>
    <h2
      className="sc-AxhCb hyOHYX"
    />
    <svg
      active="false"
      className="sc-AxgMl dCMMYF"
      onClick={[Function]}
    >
      filter.svg
    </svg>
  </section>
</section>
`;
